src/app/components/Navbar.tsx

```diff
@@ -93,10 +122,41 @@ const Navbar = () => {
   );
 };
 
-export default function NavbarComponent() {
+export default Navbar;
+
+const NavHeading = () => {
+  const pathname = usePathname();
```

Comment:
The same goes here, this side affect's value should come from the parent component.